src/H5T.c

```diff
@@ -801,7 +759,10 @@ H5T__init_package(void)
         TRUE; /* Flag to indicate whether datatype was copied or allocated (for error cleanup) */
     herr_t ret_value = SUCCEED; /* Return value */
 
-    FUNC_ENTER_NOAPI_NOINIT
+    FUNC_ENTER_NOAPI(FAIL)
+
+    if (H5_TERM_GLOBAL)
+        HGOTO_DONE(SUCCEED)
```

Comment:
Why does this one has the check for H5_TERM_GLOBAL, it didn't seem like the other packages needed it

Reply:
I will have another look. It may be some oddity that I copied from the previous code without dwelling on it, or it may be something that slipped in during a previous iteration of the patch.